src/utils/nonceManager.ts

```diff
@@ -0,0 +1,128 @@
+import type { Address } from 'abitype'
+
+import { getTransactionCount } from '../actions/public/getTransactionCount.js'
+import type { Client } from '../clients/createClient.js'
+import type { MaybePromise } from '../types/utils.js'
+import { LruMap } from './lru.js'
+
+export type CreateNonceManagerParameters = {
+  source: NonceManagerSource
+}
+
+type FunctionParameters = {
+  address: Address
+  chainId: number
+}
+
+export type NonceManager = {
+  /** Get and increment a nonce. */
+  consume(parameters: FunctionParameters & { client: Client }): Promise<number>
+  /** Increment a nonce. */
+  increment(chainId: FunctionParameters): Promise<void>
+  /** Get a nonce. */
+  get(chainId: FunctionParameters & { client: Client }): Promise<number>
+  /** Reset a nonce. */
+  reset(chainId: FunctionParameters): Promise<void>
```

Comment:
Trying to [use object property syntax instead of shorthand syntax](https://www.totaltypescript.com/method-shorthand-syntax-considered-harmful) whenever possible these days.

```suggestion
  /** Get and increment a nonce. */
  consume: (parameters: FunctionParameters & { client: Client }) => Promise<number>
  /** Increment a nonce. */
  increment: (chainId: FunctionParameters) => Promise<void>
  /** Get a nonce. */
  get: (chainId: FunctionParameters & { client: Client }) => Promise<number>
  /** Reset a nonce. */
  reset: (chainId: FunctionParameters) => Promise<void>
```